test/integration/sanity/content-type-in-form-data.test.js

```diff
@@ -0,0 +1,240 @@
+    function parseRaw(raw) {
+        raw = raw.split('\r\n');
+        var boundary = raw[0],
+            result = [],
+            data,
+            match,
+            i,
+            ii;
+
+        for (i = 0, ii = raw.length; i < ii; i++) {
+            if (raw[i] !== boundary) { continue; }
+
+            data = {};
+            match = (/\sname="(.*?)"/).exec(raw[++i]);
+            match && (data.name = match[1]);
+
+            match = (/^Content-Type:\s(.*)$/).exec(raw[++i]);
+            match && (data.contentType = match[1]);
+
+            Object.keys(data).length && result.push(data);
+        }
+
+        return result;
+    }
+
+    before(function (done) {
+        server = http.createServer(function (req, res) {
+            var rawBody = '';
+
+            req.on('data', function (chunk) {
+                rawBody += chunk.toString(); // decode buffer to string
+            }).on('end', function () {
+                res.writeHead(200, {
+                    'Content-Type': 'application/json',
+                    'Connection': 'close'
+                });
+                res.end(JSON.stringify(parseRaw(rawBody)));
+            });
+        }).listen(5050, done);
+    });
+
+    after(function (done) {
+        server.close(done);
+    });
+
+    describe('default', function () {
+        before(function (done) {
+            this.run({
+                fileResolver: fs,
+                collection: {
+                    item: [{
+                        request: {
+                            url: 'http://localhost:5050/',
+                            method: 'POST',
+                            body: {
+                                mode: 'formdata',
+                                formdata: [{
+                                    key: 'fileName',
+                                    src: 'test/fixtures/upload-csv',
+                                    type: 'file'
+                                }]
+                            }
+                        }
+                    }]
+                }
+            }, function (err, results) {
+                testrun = results;
+                done(err);
+            });
+        });
+
+        it('should complete the run', function () {
+            expect(testrun).be.ok();
+            sinon.assert.calledOnce(testrun.start);
+            sinon.assert.calledOnce(testrun.done);
+            sinon.assert.calledWith(testrun.done.getCall(0), null);
+        });
+
+        it('should upload the file without content-type in formdata', function () {
+            var request = testrun.request.getCall(0).args[3],
+                response = testrun.request.getCall(0).args[2].stream.toString();
+
+            sinon.assert.calledOnce(testrun.request);
+            sinon.assert.calledWith(testrun.request.getCall(0), null);
+
+            // content-type sent to server in request
+            expect(request.body.formdata.members[0]).to.have.property('contentType', undefined);
```

Comment:
Shouldn't we try and infer the content-type when missing?

Reply:
No, this is our data-structure, actual content-type will be inferred via the client if not specified explicitly.